test/tests/models/tx/Test_Kernel.cpp

```diff
@@ -276,4 +276,84 @@ TEST_CASE("Height-Locked")
     }
 }
 
-// TODO: Test unknown Kernel to ensure we can support soft-forking futre kernel types.
\ No newline at end of file
+TEST_CASE("Unknown Kernel")
+{
+    uint8_t features = 99;
+    uint64_t fee = 1000;
+    Commitment excess(Random::CSPRNG<33>().GetBigInt());
+    Signature signature(Random::CSPRNG<64>().GetBigInt());
+    std::vector<uint8_t> extraData = { 1, 2, 3 };
+
+    Kernel kernel(
+        features,     // Unknown type
+        fee,
+        0,
+        0,
+        boost::none,
+        std::vector<uint8_t>(extraData),
+        Commitment(excess),
+        Signature(signature)
+    );
+
+    //
+    // Serialization
+    //
+    {
+        std::vector<uint8_t> serialized = kernel.Serialized();
+        REQUIRE(serialized.size() == 110);
+
+        Deserializer deserializer(serialized);
+        REQUIRE(deserializer.Read<uint8_t>() == features);
+        REQUIRE(deserializer.Read<uint64_t>() == fee);
+        REQUIRE(deserializer.Read<uint8_t>() == extraData.size());
+        REQUIRE(deserializer.ReadVector(extraData.size()) == extraData);
+        REQUIRE(Commitment::Deserialize(deserializer) == excess);
+        REQUIRE(Signature::Deserialize(deserializer) == signature);
+
+        Deserializer deserializer2(serialized);
+        REQUIRE(kernel == Kernel::Deserialize(deserializer2));
+    }
+
+    //
+    // JSON
+    //
+    {
+        Json json(kernel.ToJSON());
+        REQUIRE(json.GetKeys() == std::vector<std::string>({ "excess", "fee", "signature", "type" }));
+        REQUIRE(json.GetRequired<Commitment>("excess") == excess);
+        REQUIRE(json.GetRequired<uint64_t>("fee") == fee);
+        REQUIRE(json.GetRequired<Signature>("signature") == signature);
+        REQUIRE(json.GetRequired<std::string>("type") == "UNKNOWN");
```

Comment:
Did we already test other types in KernelType?

Reply:
Yes. You can see the whole file with the changes here: https://github.com/ltc-mweb/libmw/blob/ab939d6cd1fe3f523f846b4ccbe17960d4e88552/test/tests/models/tx/Test_Kernel.cpp